src/google-maps/map-ground-overlay/map-ground-overlay.ts

```diff
@@ -77,21 +86,30 @@ export class MapGroundOverlay implements OnInit, OnDestroy {
   constructor(private readonly _map: GoogleMap, private readonly _ngZone: NgZone) {}
 
   ngOnInit() {
-    if (!this.bounds && (typeof ngDevMode === 'undefined' || ngDevMode)) {
-      throw Error('Image bounds are required');
-    }
     if (this._map._isBrowser) {
-      this._combineOptions().pipe(take(1)).subscribe(options => {
+      // The ground overlay setup is slightly different from the other Google Maps objects in that
+      // we have to recreate the `GroundOverlay` object whenever the bounds change, because
+      // Google Maps doesn't provide an API to update the bounds of an existing overlay.
+      this._bounds.pipe(takeUntil(this._destroyed)).subscribe(bounds => {
+        if (this.groundOverlay) {
+          this.groundOverlay.setMap(null);
+          this.groundOverlay = undefined;
+        }
+
         // Create the object outside the zone so its events don't trigger change detection.
         // We'll bring it back in inside the `MapEventManager` only for the events that the
         // user has subscribed to.
-        this._ngZone.runOutsideAngular(() => {
-          this.groundOverlay =
-              new google.maps.GroundOverlay(this._url.getValue(), this.bounds, options);
-        });
-        this._assertInitialized();
-        this.groundOverlay.setMap(this._map.googleMap!);
-        this._eventManager.setTarget(this.groundOverlay);
+        if (bounds) {
+          this._ngZone.runOutsideAngular(() => {
+            this.groundOverlay = new google.maps.GroundOverlay(this._url.getValue(), bounds, {
+              clickable: this.clickable,
+              opacity: this._opacity.value,
+            });
+          });
+          this._assertInitialized();
+          this.groundOverlay.setMap(this._map.googleMap!);
+          this._eventManager.setTarget(this.groundOverlay);
```

Comment:
Will the clear out any event listeners? Should we test that events still work on reconstituted overlays?

Reply:
It will remove the listeners, but now that I'm looking at it again, it seems like they won't be transferred to the new target. We probably haven't hit it until now, because we haven't had a case where we re-initialize a map object. This seems like it should be fixed in a separate PR since it could affect the other map components.

Reply:
I think we should hold off on submitting this pr then until the other one fixing the event manager goes through. We don't want changing the bounds to break someone's click handler

Reply:
Sure, marking as blocked.